src/chromium/crPage.ts

```diff
@@ -602,8 +591,31 @@ class FrameSession {
   }
 
   _onDetachedFromTarget(event: Protocol.Target.detachedFromTargetPayload) {
-    this._crPage.removeFrameSession(event.targetId!);
+    // This might be a worker...
     this._page._removeWorker(event.sessionId);
+
+    // ... or an oopif.
+    const childFrameSession = this._crPage._sessions.get(event.targetId!);
+    if (!childFrameSession)
+      return;
+
+    // Usually, we get frameAttached in this session first and mark child as swappedIn.
+    if (childFrameSession._swappedIn) {
+      childFrameSession.dispose();
+      return;
+    }
+
+    // However, sometimes we get detachedFromTarget before frameAttached.
+    // In this case we don't know wheter this is a remote frame detach,
+    // or just a remote -> local transition. In the latter case, frameAttached
+    // is already inflight, so let's wait for the next message. This is probably still flaky.
+    this._client._afterNextEvent(() => {
```

Comment:
Can we send a noop message to the browser and wait for its roundtrip to be sure that we don't miss possible `frameAttached` which is already inflight?

Better yet would be to make it deterministic in Chrome but I assume it's non-trivial :grin:

Reply:
That's a great idea! Sending `Page.enable` now and it helps.

Fixing in Chrome sounds hard 😞